src/type.ts

```diff
@@ -28,10 +41,10 @@ export const TYPE_FROM_SHORT_TYPE = {
 
 /**
  * Get full, lowercase type name for a given type.
- * @param  {String} type
- * @return {String} Full type name.
+ * @param  type
+ * @return Full type name.
  */
-export function getFullName(type: string): string {
-  return TYPE_FROM_SHORT_TYPE[type.toUpperCase()] || // short type is uppercase by default
-         type.toLowerCase();
+export function getFullName(type: Type): Type {
+  return TYPE_FROM_SHORT_TYPE[Type[type].toUpperCase()] || // short type is uppercase by default
```

Comment:
I read the code and also doesn’t understand why it works for short type and from pulling and testing — this does not work for short type!

I found this extremely confusing and if the one who proposes it also get confused by it, I don’t think this is worth it.